### PR TITLE
Group together all sections related to i18n in dev guide

### PR DESCRIPTION
**Pull request by Hyzual**
As a Tuleap developper, most of the time I am switching between back-end
and front-end work. It was hard to find the relevant doc section for
each context. Now, they are all grouped together so we have a common
place to explain "how to translate a string" in tuleap.

### CODE REVIEW

File: languages/en/developer-guide/internationalization.rst [hunk 1]
```diff
@@ -0,0 +1,28 @@
+.. _i18n:
+
+Internationalization (i18n)
+===========================
+
+i18n in Tuleap should always be done with gettext. Depending on where you are,
+you should use different means to translate your strings. One thing is constant
```
Comment (magarnier) [inline languages/en/developer-guide/internationalization.rst]: s/means/ways

means to seems weird to me

File: languages/en/developer-guide/i18n/back-end.rst [hunk 2]
```diff
@@ -30,12 +33,14 @@ In REST routes
...
-  
-     throw new RestException(
-           404,
-           null,
-           ['i18n_error_message' => dgettext('tuleap-docman', 'The folder does not exist.')]
-     );
+
+     throw new I18NRestException(
+          400,
+          sprintf(
```
Comment (magarnier) [inline languages/en/developer-guide/i18n/back-end.rst]: Is it expected that you removed the key i18n_error_message?

File: languages/en/developer-guide/internationalization.rst (hunk 1 above)
Comment (Hyzual) [reply languages/en/developer-guide/internationalization.rst]: "means" => moyens.
I'm okay with changing it.

File: languages/en/developer-guide/i18n/back-end.rst (hunk 2 above)
Comment (Hyzual) [reply languages/en/developer-guide/i18n/back-end.rst]: Yes, I18NRestException does this for us.

**Pull request merged**